File: client/src/app/entities/supplierpayment.ts
import {User} from './user';
import {Deduction} from './deduction';
import {Collection} from './collection';
import {Paymenttype} from './paymenttype';
import {DataPage} from '../shared/data-page';
import {Paymentstatus} from './paymentstatus';

export class Supplierpayment {
  id: number;
  code: string;
  date: string;
  amount: number;
  chequeno: string;
  chequebank: string;
  chequebranch: string;
  chequedate: string;
  paymenttype: Paymenttype;
  paymentstatus: Paymentstatus;
  collection: Collection;
  description: string;
  deductionList: Deduction[];
  creator: User;
  tocreation: string;
}

export class SupplierpaymentDataPage extends DataPage{
    content: Supplierpayment[];
}

File: server/src/main/java/bit/project/server/entity/Supplierpayment.java
package bit.project.server.entity;

import lombok.Data;
import java.util.List;
import java.time.LocalDate;
import javax.persistence.*;
import javax.persistence.Id;
import java.math.BigDecimal;
import javax.persistence.Lob;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Supplierpayment{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate date;

    private BigDecimal amount;

    private String chequeno;

    private String chequebank;

    private String chequebranch;

    private LocalDate chequedate;

    @Lob
    private String description;

    private LocalDateTime tocreation;


    @ManyToOne
    private Paymenttype paymenttype;

    @ManyToOne
    private Paymentstatus paymentstatus;

    @ManyToOne
    private Collection collection;

    @OneToMany(mappedBy="supplierpayment", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Deduction> deductionList;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Supplierpayment(Integer id){
        this.id = id;
    }

    public Supplierpayment(Integer id, String code){
        this.id = id;
        this.code = code;
    }

}